### DISCUSSION TITLE
Is SvelteKit mature enough for production application?

### DISCUSSION THREAD

**Discussion by shahensaifullah** [sveltejs/kit#6203] [Q&A]
### My team will build an eCommerce website for a client. We are a Vue people especially using Nuxt. but we think we should use svelte for development. Should we use Sveltekit for development?

**Comment by nikfp** [comment]
I would say yes, BUT.... 

Remember that Sveltekit is still in beta. The API could change and if your team got high dev velocity and then a breaking change hit, you would be in pretty bad shape. Recently there was a big breaking change related to routing structure and file naming conventions that has the community a bit divided. I have my opinion of it (positive), others, not so much. I do know that Sveltekit is close to 1.0 though. And I also know that the dev experience is really, really good. 

Also consider what you need Sveltekit to do for you. If you are building something that could run as a SPA you might look at vanilla Svelte and the ecosystem surrounding it. The reason I say this is that the API will be stable. On the flipside, if you need Sveltekit features, you can keep tabs on the SK discussions and you'll see any breaking changes coming. It's just more to stay on top of. 

So I would say that if you can be tolerant of some potential breaking changes and having to do some refactors as a result, start with Sveltekit now and don't look back. People give Svelte and Sveltekit a lot of love for good reason.

**Reply by shahensaifullah** [reply]
I also love Svelte and also its staff. but big chances like recent file naming routing change, or removed session can create big project problems. 
Do you know when it will be in a stable version?

**Reply by richarddavenport** [reply]
Do you already have a big project? If not, but you will be building a "big" project, chances are you won't move as fast as you think you will. If SvelteKit changes something, most likely you will be able to handle the migration in under a day. Especially if you think you'll be able to build a "big" site as fast as you think you.

**Reply by acoyfellow** [reply]
As someone who has been gambling for over a year on SK in production (more than one b2b SaaS)...

Don't do it, yet. 

If you had asked me this before the latest routing, + file convention, and session updates, I'd likely have said yes. 

I've got 150+ files in my routes folder, and I really leaned into the old style routing, single page file, and old endpoints. I'm not migrating that project until the dust settles, maybe even 1.0 comes out.

**Reply by 0gust1** [reply]
similar situation as @acoyfellow here (dozen of b2b SaaS instances), although we have half-less impacted files (~70 ish)

Just after running the migration tool, I felt a "little tired" seeing the breadth of changes... haha. 
But today I almost finished our migration (10 files left). It was somewhat quicker than expected (but still cumbersome). 
On the positive side, we now much better understand the new router system and all the implications of choices made by maintainers, and we could delete a lot of boilerplate type-related code thanks to the tighter TS integration.

IDK if it helps, but keep your spirits up!

**Comment by madeleineostoja** [comment]
My answer as an unmitigated fangirl of Svelte and Sveltekit: Hell no.

I'm using svelte(kit) all the way down on a very large platform, and it's wonderful, but I'm also there to maintain and continually refactor it as breaking changes come. And they do come, with increasing frequency. The stability of the framework itself is pretty solid, but the stability of the API design has been extremely volatile.

I think handing over a sveltekit project to a client now would be very irresponsible, unless you're on retainer to maintain it, at least until Sveltekit hits v1

**Reply by shahensaifullah** [reply]
When will release v1? is there any roadmap?

**Reply by madeleineostoja** [reply]
when it's ready https://github.com/sveltejs/kit/milestone/2

**Comment by jesseskinner** [comment]
In my opinion, if I'm starting a brand new application or website that is going to be developed over the long term, I would rather go with SvelteKit today than pick something that I'm going to regret in a couple of years.

There has been some significant maintenance overhead over the past year, and it looks like the lead up to v1.0 will likely see further changes. I still feel like the time savings of using such an elegant language and framework outweigh the time spent migrating to new releases. The recent major routing change impacted a lot of files, but the migration tool is so good, migrating a web app with 50 routes including a server-side API took me only half an hour.

To avoid the ongoing maintenance, you could anchor the project down to today's version of SvelteKit, and then do one big migration when v1 comes out, at your convenience. I have faith that SvelteKit will stabilize drastically once v1 comes out, like Svelte did when v3 came out.

**Comment by danawoodman** [comment]
Only use if you're ok with some major changes before v1. If you're ok with managing regular migrations and you plan for that, do it, otherwise I'd just wait till 1.0

**Reply by shahensaifullah** [reply]
waiting is the best decition for that.

**Comment by 0gust1** [comment]
> mature enough for production application

It depends (on your capabilities/experience/willingness). It's a decision you have to assume. 
We have a production app with sveltekit since 1.5 year, when I made the decision, I knew where I was going into and I knew it could be a bumpy ride sometimes.

> My team will build an eCommerce website for a client.

Seems risky/unprofessional if you have to hand over the website/app and close the contract before SK hits 1.0.0.

>We are a Vue people especially using Nuxt. but we think we should use svelte for development. Should we use Sveltekit for development

Don't forget you can still use Svelte alone. Svelte itself is rock solid, and very stable. Yes, you would have to use/choose something for the backend, and find a way to send working HTML from your server.

I empathize with your dilemna (been there, done that :) ), IMHO, it mostly depends on the time-frame of your project/contract, and your willingness to assume potential breaking changes before 1.0.0. (watching the maintainers working since 1.5 year, I would say that most of the big breaking ones are behind... but who knows ?)
